src/FirmwarePlugin/APM/ArduCopterFirmwarePlugin.cc

```diff
@@ -136,3 +136,44 @@ bool ArduCopterFirmwarePlugin::vehicleYawsToNextWaypointInMission(const Vehicle*
     return true;
 }
 
+void ArduCopterFirmwarePlugin::guidedModeGotoLocation(Vehicle* vehicle, const QGeoCoordinate& gotoCoord)
+{
+    if (qIsNaN(vehicle->altitudeRelative()->rawValue().toDouble())) {
+        qgcApp()->showMessage(QStringLiteral("Unable to go to location, vehicle position not known."));
+        return;
+    }
+
+    setGuidedMode(vehicle, true);
+
+    if (vehicle->capabilityBits() & MAV_PROTOCOL_CAPABILITY_COMMAND_INT) {
+
+        mavlink_message_t msg;
+        mavlink_set_position_target_global_int_t cmd;
```

Comment:
Per discussion about MAVLink confusion; maybe rename this, since it isn't a Command that you're sending?